File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task34_CLAUDE_claude-3-5-sonnet-20240620.java

public class NarcissisticNumber {
    public static boolean isNarcissistic(int number) {
        String numStr = String.valueOf(number);
        int length = numStr.length();
        int sum = 0;
        
        for (char digit : numStr.toCharArray()) {
            sum += Math.pow(Character.getNumericValue(digit), length);
        }
        
        return sum == number;
    }

    public static void main(String[] args) {
        System.out.println(isNarcissistic(153));  // true
        System.out.println(isNarcissistic(1652)); // false
    }
}

File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task34_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <cmath>
#include <string>

bool isNarcissistic(int number) {
    std::string numStr = std::to_string(number);
    int length = numStr.length();
    int sum = 0;

    for (char digit : numStr) {
        sum += std::pow(digit - \'0\', length);
    }

    return sum == number;
}

int main() {
    std::cout << std::boolalpha;
    std::cout << isNarcissistic(153) << std::endl;  // true
    std::cout << isNarcissistic(1652) << std::endl; // false
    return 0;
}
